### ISSUE TITLE
Missing documentation on overlay port behaviour (overrides related)

### ISSUE THREAD

**Issue by bytebunny** [microsoft/vcpkg#37478]
**Describe the bug**
The [overlay ports documentation](https://learn.microsoft.com/en-us/vcpkg/concepts/overlay-ports) lacks description of their behaviour when used in combination with [overrides](https://learn.microsoft.com/en-us/vcpkg/users/versioning#overrides), which in its turn does not refer to overlay ports either.

The aforementioned behaviour is the following. When `overlay-ports` field is used, the information stated in `overrides` field is ignored.
This turns out to be documented in the Microsoft [blog post](https://devblogs.microsoft.com/cppblog/take-control-of-your-vcpkg-dependencies-with-versioning-support/):
![vcpkg-overlay-with-overrides](https://github.com/microsoft/vcpkg/assets/4153521/bd7f848c-dd1f-4735-b976-cec30f895233)

What this paragraph entails is that when one has a manifest file with `overrides` field stating that version `x.y.z` of the package is used, that may not at all be the case, because only the version stated in the `vcpkg.json` file of the package under `ports` is considered.

I reckon it is a bug in the documentation as it is not reasonable to expect from a user of VCPKG reading the docs to stumble upon that blog post, so they would have to find out about this behaviour the hard way. And I believe that people reading the `vcpkg.json` might misunderstand which version will actually be installed under these circumstances.

It would be useful if VCPKG warned the user that they have an `overrides` field, but that it will be ignored because of the presence of `overlay-ports`. This way, even people not reading the documentation carefully, would have the means of inferring knowledge about this behaviour.

**Environment**
- Microsoft VCPKG documentation as of mid March 2024.

**To Reproduce**
1. Add `overrides` field to `vcpkg.json`:
    ```
    {
      "name": "project",
      "overrides": [
         { "name": "some-package", "version": "x.y.z" }
      ]
    }
    ```
2. Add `overlay-ports` field to `vcpkg-configuration.json`:
    ```
    {
      "default-registry": {
       ...
      },
      "overlay-ports": [
         "<path/to/ports>",
      ]
    }
    ```
3. Add port for `some-package` under `<path/to/ports>` with the package version different from `x.y.z`
4. Install "some-package"
5. Lo and behold that the installed version is **not** `x.y.z`
 
**Expected behavior**
During the installation a warning is issued by VCPKG reporting that the version installed might be different from the expected.

**Failure logs**
N/A

**Comment by Neumann-A** [comment]
> While [resolving package names](https://learn.microsoft.com/en-us/vcpkg/concepts/package-name-resolution), overlay ports take priority.

from https://learn.microsoft.com/en-us/vcpkg/concepts/overlay-ports

Basic rule:

overlay > overrides > min version constrain > baseline

**Comment by bytebunny** [comment]
@Neumann-A I don't agree that mentioning priority rule for package **name** resolution would be read by users as "when you have both `overlay-ports` and `overrides` in your manifest, the `overrides` aren't overriding **version**s".

**Comment by Neumann-A** [comment]
You need to first resolve the name before you can even think about version resolution. Since overlays only provide one version that version will be selected.

**Comment by bytebunny** [comment]
I still think the existence of "Versions and custom ports" section in the blog speaks in favour of my argument. I also think that a warning should be issued by the tool, since people might be reading the manifest without consulting the documentation.

That being said, I am not here to insist on anything. Feel free to close this issue.
